### ISSUE TITLE
Cannot start Jackett: System.BadImageFormatException: Method has no body

### ISSUE THREAD

**Issue by truenicoco** [Jackett/Jackett#4531]
**Jackett version**:  v0.10.657.0
**Mono version** (if not using Windows): 5.18.0.240 (Debian 5.18.0.240+dfsg-1 Fri Jan 18 17:30:45 UTC 2019)

I don't know much about Mono but something happened during my latest `apt upgrade` on my debian buster (testing) box.

I tried downloading the jackett binary to make a clean install but it didn't solve the problem. Please let me know if can I provide additional details.

```
$ /usr/bin/mono /opt/Jackett/JackettConsole.exe --NoRestart
01-24 13:38:33 Info Starting Jackett v0.10.657.0 
01-24 13:38:33 Error System.InvalidOperationException: object_op ---> System.BadImageFormatException: Method has no body
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEvents () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                                                    
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEventsIfNotDisposed () [0x0001c] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                                       
  at System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x0002a] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                           
  at (wrapper remoting-invoke-with-check) System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents(bool)                                                                         
  at System.IO.CoreFXFileSystemWatcherProxy.<StartDispatching>m__0 (System.IO.CoreFX.FileSystemWatcher internal_fsw, System.IO.FileSystemWatcher fsw) [0x00048] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                                                                                                                                               
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001d3] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                                                                                                               
   --- End of inner exception stack trace ---                                                                                                                                      
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001f8] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                                                                                                               
  at System.IO.CoreFXFileSystemWatcherProxy.StartDispatching (System.Object handle) [0x00007] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                              
  at System.IO.FileSystemWatcher.Start () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                                                                        
  at System.IO.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x00049] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                                  
  at (wrapper remoting-invoke-with-check) System.IO.FileSystemWatcher.set_EnableRaisingEvents(bool)                                                                                
  at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.TryEnableFileSystemWatcher () [0x00038] in <09eece63ae304250b9ea1428a3dd9267>:0                              
  at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.CreateFileChangeToken (System.String filter) [0x00033] in <09eece63ae304250b9ea1428a3dd9267>:0               
  at Microsoft.Extensions.FileProviders.PhysicalFileProvider.Watch (System.String filter) [0x00024] in <09eece63ae304250b9ea1428a3dd9267>:0                                        
  at Microsoft.Extensions.Configuration.FileConfigurationProvider.<.ctor>b__0_0 () [0x00016] in <797c96f788a84ce18b807cacba80ff0d>:0                                               
  at Microsoft.Extensions.Primitives.ChangeToken.OnChange (System.Func`1[TResult] changeTokenProducer, System.Action changeTokenConsumer) [0x00053] in <298b1e1086f046869486e3dc04d2ecef>:0                                                                                                                                                                           
  at Microsoft.Extensions.Configuration.FileConfigurationProvider..ctor (Microsoft.Extensions.Configuration.FileConfigurationSource source) [0x00035] in <797c96f788a84ce18b807cacba80ff0d>:0                                                                                                                                                                         
  at Microsoft.Extensions.Configuration.Json.JsonConfigurationProvider..ctor (Microsoft.Extensions.Configuration.Json.JsonConfigurationSource source) [0x00000] in <d95d925454834fb7b5eca81f4ca54bd3>:0                                                                                                                                                               
  at Microsoft.Extensions.Configuration.Json.JsonConfigurationSource.Build (Microsoft.Extensions.Configuration.IConfigurationBuilder builder) [0x00007] in <d95d925454834fb7b5eca81f4ca54bd3>:0                                                                                                                                                                       
  at Microsoft.Extensions.Configuration.ConfigurationBuilder.Build () [0x0001a] in <83ce78fdf0ae48fcb73903e2539736de>:0                                                            
  at Microsoft.AspNetCore.Hosting.WebHostBuilder.BuildCommonServices (System.AggregateException& hostingStartupErrors) [0x001d7] in <127e718ac42e4542a71b54cec3ef156b>:0           
  at Microsoft.AspNetCore.Hosting.WebHostBuilder.Build () [0x0001a] in <127e718ac42e4542a71b54cec3ef156b>:0                                                                        
  at Jackett.Server.Program.Main (System.String[] args) [0x002c1] in <7e13cdf66d91425ebc181b4761a7b728>:0  System.InvalidOperationException: object_op ---> System.BadImageFormatException: Method has no body                                                                                                                                                        
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEvents () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                                                    
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEventsIfNotDisposed () [0x0001c] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                                       
  at System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x0002a] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                           
  at (wrapper remoting-invoke-with-check) System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents(bool)                                                                         
  at System.IO.CoreFXFileSystemWatcherProxy.<StartDispatching>m__0 (System.IO.CoreFX.FileSystemWatcher internal_fsw, System.IO.FileSystemWatcher fsw) [0x00048] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                                                                                                                                               
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001d3] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                                                                                                               
   --- End of inner exception stack trace ---                                                                                                                                      
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001f8] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                                                                                                               
  at System.IO.CoreFXFileSystemWatcherProxy.StartDispatching (System.Object handle) [0x00007] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                              
  at System.IO.FileSystemWatcher.Start () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                                                                        
  at System.IO.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x00049] in <a9a08e39ba304bd0a84c49bd158dfc02>:0                                                  
  at (wrapper remoting-invoke-with-check) System.IO.FileSystemWatcher.set_EnableRaisingEvents(bool)                                                                                
  at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.TryEnableFileSystemWatcher () [0x00038] in <09eece63ae304250b9ea1428a3dd9267>:0                              
  at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.CreateFileChangeToken (System.String filter) [0x00033] in <09eece63ae304250b9ea1428a3dd9267>:0               
  at Microsoft.Extensions.FileProviders.PhysicalFileProvider.Watch (System.String filter) [0x00024] in <09eece63ae304250b9ea1428a3dd9267>:0                                        
  at Microsoft.Extensions.Configuration.FileConfigurationProvider.<.ctor>b__0_0 () [0x00016] in <797c96f788a84ce18b807cacba80ff0d>:0                                               
  at Microsoft.Extensions.Primitives.ChangeToken.OnChange (System.Func`1[TResult] changeTokenProducer, System.Action changeTokenConsumer) [0x00053] in <298b1e1086f046869486e3dc04d2ecef>:0                                                                                                                                                                           
  at Microsoft.Extensions.Configuration.FileConfigurationProvider..ctor (Microsoft.Extensions.Configuration.FileConfigurationSource source) [0x00035] in <797c96f788a84ce18b807cacba80ff0d>:0                                                                                                                                                                         
  at Microsoft.Extensions.Configuration.Json.JsonConfigurationProvider..ctor (Microsoft.Extensions.Configuration.Json.JsonConfigurationSource source) [0x00000] in <d95d925454834fb7b5eca81f4ca54bd3>:0                                                                                                                                                               
  at Microsoft.Extensions.Configuration.Json.JsonConfigurationSource.Build (Microsoft.Extensions.Configuration.IConfigurationBuilder builder) [0x00007] in <d95d925454834fb7b5eca81f4ca54bd3>:0                                                                                                                                                                       
  at Microsoft.Extensions.Configuration.ConfigurationBuilder.Build () [0x0001a] in <83ce78fdf0ae48fcb73903e2539736de>:0                                                            
  at Microsoft.AspNetCore.Hosting.WebHostBuilder.BuildCommonServices (System.AggregateException& hostingStartupErrors) [0x001d7] in <127e718ac42e4542a71b54cec3ef156b>:0           
  at Microsoft.AspNetCore.Hosting.WebHostBuilder.Build () [0x0001a] in <127e718ac42e4542a71b54cec3ef156b>:0                                                                        
  at Jackett.Server.Program.Main (System.String[] args) [0x002c1] in <7e13cdf66d91425ebc181b4761a7b728>:0 

Unhandled Exception:
System.InvalidOperationException: object_op ---> System.BadImageFormatException: Method has no body
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEvents () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEventsIfNotDisposed () [0x0001c] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x0002a] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at (wrapper remoting-invoke-with-check) System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at System.IO.CoreFXFileSystemWatcherProxy.<StartDispatching>m__0 (System.IO.CoreFX.FileSystemWatcher internal_fsw, System.IO.FileSystemWatcher fsw) [0x00048] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001d3] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
   --- End of inner exception stack trace ---
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001f8] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFXFileSystemWatcherProxy.StartDispatching (System.Object handle) [0x00007] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.FileSystemWatcher.Start () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x00049] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.TryEnableFileSystemWatcher () [0x00038] in <09eece63ae304250b9ea1428a3dd9267>:0 
  at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.CreateFileChangeToken (System.String filter) [0x00033] in <09eece63ae304250b9ea1428a3dd9267>:0 
  at Microsoft.Extensions.FileProviders.PhysicalFileProvider.Watch (System.String filter) [0x00024] in <09eece63ae304250b9ea1428a3dd9267>:0 
  at Microsoft.Extensions.Configuration.FileConfigurationProvider.<.ctor>b__0_0 () [0x00016] in <797c96f788a84ce18b807cacba80ff0d>:0 
  at Microsoft.Extensions.Primitives.ChangeToken.OnChange (System.Func`1[TResult] changeTokenProducer, System.Action changeTokenConsumer) [0x00053] in <298b1e1086f046869486e3dc04d2ecef>:0 
  at Microsoft.Extensions.Configuration.FileConfigurationProvider..ctor (Microsoft.Extensions.Configuration.FileConfigurationSource source) [0x00035] in <797c96f788a84ce18b807cacba80ff0d>:0 
  at Microsoft.Extensions.Configuration.Json.JsonConfigurationProvider..ctor (Microsoft.Extensions.Configuration.Json.JsonConfigurationSource source) [0x00000] in <d95d925454834fb7b5eca81f4ca54bd3>:0 
  at Microsoft.Extensions.Configuration.Json.JsonConfigurationSource.Build (Microsoft.Extensions.Configuration.IConfigurationBuilder builder) [0x00007] in <d95d925454834fb7b5eca81f4ca54bd3>:0 
  at Microsoft.Extensions.Configuration.ConfigurationBuilder.Build () [0x0001a] in <83ce78fdf0ae48fcb73903e2539736de>:0 
  at Microsoft.AspNetCore.Hosting.WebHostBuilder.BuildCommonServices (System.AggregateException& hostingStartupErrors) [0x001d7] in <127e718ac42e4542a71b54cec3ef156b>:0 
  at Microsoft.AspNetCore.Hosting.WebHostBuilder.Build () [0x0001a] in <127e718ac42e4542a71b54cec3ef156b>:0 
  at Jackett.Server.Program.Main (System.String[] args) [0x00302] in <7e13cdf66d91425ebc181b4761a7b728>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: object_op ---> System.BadImageFormatException: Method has no body
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEvents () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEventsIfNotDisposed () [0x0001c] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x0002a] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at (wrapper remoting-invoke-with-check) System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at System.IO.CoreFXFileSystemWatcherProxy.<StartDispatching>m__0 (System.IO.CoreFX.FileSystemWatcher internal_fsw, System.IO.FileSystemWatcher fsw) [0x00048] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001d3] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
   --- End of inner exception stack trace ---
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001f8] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFXFileSystemWatcherProxy.StartDispatching (System.Object handle) [0x00007] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.FileSystemWatcher.Start () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x00049] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.TryEnableFileSystemWatcher () [0x00038] in <09eece63ae304250b9ea1428a3dd9267>:0 
  at Microsoft.Extensions.FileProviders.Physical.PhysicalFilesWatcher.CreateFileChangeToken (System.String filter) [0x00033] in <09eece63ae304250b9ea1428a3dd9267>:0 
  at Microsoft.Extensions.FileProviders.PhysicalFileProvider.Watch (System.String filter) [0x00024] in <09eece63ae304250b9ea1428a3dd9267>:0 
  at Microsoft.Extensions.Configuration.FileConfigurationProvider.<.ctor>b__0_0 () [0x00016] in <797c96f788a84ce18b807cacba80ff0d>:0 
  at Microsoft.Extensions.Primitives.ChangeToken.OnChange (System.Func`1[TResult] changeTokenProducer, System.Action changeTokenConsumer) [0x00053] in <298b1e1086f046869486e3dc04d2ecef>:0 
  at Microsoft.Extensions.Configuration.FileConfigurationProvider..ctor (Microsoft.Extensions.Configuration.FileConfigurationSource source) [0x00035] in <797c96f788a84ce18b807cacba80ff0d>:0 
  at Microsoft.Extensions.Configuration.Json.JsonConfigurationProvider..ctor (Microsoft.Extensions.Configuration.Json.JsonConfigurationSource source) [0x00000] in <d95d925454834fb7b5eca81f4ca54bd3>:0 
  at Microsoft.Extensions.Configuration.Json.JsonConfigurationSource.Build (Microsoft.Extensions.Configuration.IConfigurationBuilder builder) [0x00007] in <d95d925454834fb7b5eca81f4ca54bd3>:0 
  at Microsoft.Extensions.Configuration.ConfigurationBuilder.Build () [0x0001a] in <83ce78fdf0ae48fcb73903e2539736de>:0 
  at Microsoft.AspNetCore.Hosting.WebHostBuilder.BuildCommonServices (System.AggregateException& hostingStartupErrors) [0x001d7] in <127e718ac42e4542a71b54cec3ef156b>:0 
  at Microsoft.AspNetCore.Hosting.WebHostBuilder.Build () [0x0001a] in <127e718ac42e4542a71b54cec3ef156b>:0 
  at Jackett.Server.Program.Main (System.String[] args) [0x00302] in <7e13cdf66d91425ebc181b4761a7b728>:0
```

**Comment by truenicoco** [comment]
FWIW, my workaround was to follow the build instructions, unfortunately I cannot use the autoupdater now…

**Comment by adamhough** [comment]
https://github.com/AArnott/Nerdbank.GitVersioning/issues/118#issuecomment-294321184 Looks like it might be project build issue though not sure if it will get fixed in this project since there appears to be a transition to the MS nodnet packages instead of Mono.

**Comment by IronTooch** [comment]
For anyone getting here from the Google for a problem with 
```
 Fatal Unable to start Kestrel. System.BadImageFormatException: Could not load file or assembly 'System.Security.Cryptography.X509Certific
ates, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The module was expected to contain an assembly manifest.
May  2 07:20:19 casino jackett[1063]: File name: 'System.Security.Cryptography.X509Certificates, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
May  2 07:20:19 casino jackett[1063]:    at Microsoft.AspNetCore.Server.Kestrel.KestrelConfigurationLoader.Load()
May  2 07:20:19 casino jackett[1063]:    at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServer.ValidateOptions()
May  2 07:20:19 casino jackett[1063]:    at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServer.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)
May  2 07:20:19 casino jackett[1063]: 05-02 07:20:19 Fatal Unable to start Kestrel. System.BadImageFormatException: Could not load file or assembly 'System.Security.Cryptography.X509Certific
ates, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The module was expected to contain an assembly manifest.
```

The answer is to make the .config and Jackett path in your user's home directory.

**Comment by frebib** [comment]
@flightlevel Why did you close this? Multiple people are hitting this issue and you didn't provide any remedy/answer

**Comment by steenbras** [comment]
The problem I had was different from the issue @IronTooch described, and I found a pointer to the workaround [here](https://github.com/mono/mono/issues/12557). 

TL;DR - check what  `libmono-native.so` binary you have in your `/usr/lib` directory and ensure that it matches what's in `/etc/mono/conf`. Either add it there or create a symlink that matches what the conf file expects.
